### ISSUE TITLE
Syntax highlight for unpackPhase/installPhase/...

### ISSUE THREAD

**Issue by Freed-Wu** [LnL7/vim-nix#54]
How about highlight unpackPhase/installPhase/... as sh syntax? Like markdown/typst highlight code fence:

![markdown](https://user-images.githubusercontent.com/32936898/230709055-10e9fc61-e2ca-417a-b060-7d156915d22e.png)

Refer <https://github.com/vim-pandoc/vim-pandoc-syntax>

**Comment by bew** [comment]
That looks like a restricted version of #28

**Comment by Ma27** [comment]
IMHO it's not a good idea to assume bash for `unpackPhase` etc. Also, this highlighting would be pretty incomplete considering that you can define custom phases, have quite a lot of phases and also pre/post hooks _and_ the syntax highlighting would be completely wrong and confusing if you have - for whatever reason - an `installPhase` that's not a bash string. To deliver somehow usable results, we'd probably have to scan for `stdenv.mkDerivation` which would make this even more error-prone (right now we simply pattern-match for some highlighting even in case of syntax errors) and we'd need some more sophisticated parsing then.

tl;dr what @bew says, the correct solution would be something like #28 (although getting this ready is a different story).

Btw just tried #28 out though and it works relatively well, so you just apply the patch to your local `vim-nix` (that's possible via `super.vimUtils.vim-nix.overrideAttrs` in an overlay for instance).
